pkg/config/node.go

```diff
@@ -430,3 +437,45 @@ func GetClusterNameAndDomain(kubeconfigPath, clusterConfigPath string) (clusterN
 
 	return
 }
+
+func PopulateNodeAddresses(kubeconfigPath string, node *Node) {
+	// Get node list
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		log.Errorf("Failed to build client config: %s", err)
+		return
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Errorf("Failed to create client: %s", err)
+		return
+	}
+	nodes, err := clientset.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("Failed to get node list: %s", err)
+		return
+	}
+	for _, n := range nodes.Items {
+		name := ""
+		addr := net.IP{}
+		for _, a := range n.Status.Addresses {
+			if a.Type == v1.NodeHostName {
+				// We only want the shortname
+				name = strings.Split(a.Address, ".")[0]
+			}
+			if a.Type == v1.NodeInternalIP {
+				addr = net.ParseIP(a.Address)
+			}
+		}
+		// TODO(bnemec): The ipv6 flag isn't currently used in the templates,
+		// but at some point it probably should be so we provide RFC-compliant
+		// ipv6 behavior.
+		ipv6 := true
+		check := addr.To4()
+		if check != nil {
+			ipv6 = false
+		}
```

Comment:
To be on the safe side, maybe we can append entry to node list only if : 
		name != = "" and 		addr != net.IP{} ?

Reply:
I feel like it would be an error if we got a node record that looked like that, but it shouldn't hurt to add a check.